client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,64 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final RetentionConfig retentionConfig;
+
+    public enum RetentionConfig {
+        /**
+         * This {@link ReaderGroup} is not a subscriber and does not participate in Consumption Based Retention.
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        NONE(false, false),
+
+        /**
+         * This {@link ReaderGroup} is a subscriber and can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream according to its read positions. These read positions have to be provided
+         * manually in the form of a {@link StreamCut}, see this method:
+         *
+         * {@link ReaderGroup#updateRetentionStreamCut(Map)}
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
```

Comment:
Do not link to PDPs in public documentation. Those are for describing the implementation for our benefit. Users should not be expected to read or understand them.

Reply:
Yeah I've removed the link.